src/Features/Core/Portable/InheritanceMargin/InheritanceMarginItem.cs

```diff
@@ -58,6 +58,19 @@ namespace Microsoft.CodeAnalysis.InheritanceMargin
             TargetItems = targetItems;
         }
 
+        public override int GetHashCode()
+            => throw ExceptionUtilities.Unreachable();
+
+        public override bool Equals(object? obj)
+            => obj is InheritanceMarginItem item && Equals(item);
+
+        public bool Equals(InheritanceMarginItem other)
+            => this.LineNumber == other.LineNumber &&
```

Comment:
@Cosifne It would be good to not have LineNumber be part of hte data side of this.  It is positional, and shouldn't cause us to teardown/recreate tags.  In other words, if a member moves down one line, we should still be able to use its existing tags insteading of having to remove hte old one and create a new one.  

teh general pattern for this is to have ITagSpan in the tagger store the position, and the ITag itself it position-free.

Reply:
Should we get a bug tracking this?